patch/field_type_test.go

```diff
@@ -0,0 +1,270 @@
+package patch
+
+import (
+	"go/ast"
+	"go/parser"
+	"go/token"
+	"go/types"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"google.golang.org/protobuf/compiler/protogen"
+)
+
+const (
```

Comment:
Could you move these `const` and `type` definitions into `prepareCastType`? These are fairly generic named types and constants, and if they don’t need to be in the global namespace, let’s keep them out.